### PR TITLE
Deprecate commands from the transformers-cli that are in the hf-cli

### PR DESCRIPTION
**Pull request by LysandreJik**
Commands that are both in the `transformers-cli` and in the `huggingface-cli` are deprecated here and will be quickly removed.

I'm voting for deprecating them and not removing them even though better ways exist as I suspect some users to use the `transformers-cli` in bash scripts to automatically upload models to the hub.

Context from @julien-c:

> my thoughts is that we should deprecate the subset of transformers-cli command that are in huggingface-cli, as the commands are identical and having both is confusing.
>
> Transformers-specific commands (model conversion, new model templating) can stay in transformers-cli.
>
> What do you think?

### CODE REVIEW

**Comment by julien-c** [comment]
my reason for removing (still keeping a descriptive error obviously) rather than deprecating is that I'd love to know if people actually use those (and if they do, do they use them in scripts or manually)

But I will defer to the great transformers-maintainers as the final decision 💖

**Review by sgugger — APPROVED** [review]
We will know for sure after people ignore the deprecation warnings and complain in v5 when we actually remove them ;-)
But we go to convoluted lengths to avoid breaking changes in situations way smaller than this, so I feel very strongly about going through the normal deprecation cycle and wait for a major release before removing.

**Comment by julien-c** [comment]
sounds good to me!

**Pull request merged**
